src/foundation.cs

```diff
@@ -10351,8 +10352,44 @@ partial interface NSItemProvider : NSCopying {
 		[MacCatalyst (13, 0)]
 		[Export ("preferredPresentationStyle", ArgumentSemantic.Assign)]
 		UIPreferredPresentationStyle PreferredPresentationStyle { get; set; }
+
+		// from interface UTType (NSItemProvider)
+
+		[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+		[Export ("initWithContentsOfURL:contentType:openInPlace:coordinated:visibility:")]
+		NativeHandle Constructor (NSUrl fileUrl, [NullAllowed] UTType contentType, bool openInPlace, bool coordinated, NSItemProviderRepresentationVisibility visibility);
+
+		[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+		[Export ("registerDataRepresentationForContentType:visibility:loadHandler:")]
+		void RegisterDataRepresentation (UTType contentType, NSItemProviderRepresentationVisibility visibility, NSItemProviderUTTypeLoadDelegate loadHandler);
+
+		[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+		[Export ("registerFileRepresentationForContentType:visibility:openInPlace:loadHandler:")]
+		void RegisterFileRepresentation (UTType contentType, NSItemProviderRepresentationVisibility visibility, bool openInPlace, NSItemProviderUTTypeLoadDelegate loadHandler);
+
+		[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+		[Export ("registeredContentTypes", ArgumentSemantic.Copy)]
+		UTType[] RegisteredContentTypes { get; }
+
+		[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+		[Export ("registeredContentTypesForOpenInPlace", ArgumentSemantic.Copy)]
+		UTType[] RegisteredContentTypesForOpenInPlace { get; }
+
+		[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+		[Export ("registeredContentTypesConformingToContentType:")]
+		UTType[] RegisteredContentTypesConforming (UTType contentType);
+
+		[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+		[Export ("loadDataRepresentationForContentType:completionHandler:")]
+		NSProgress LoadDataRepresentation (UTType contentType, ItemProviderDataCompletionHandler completionHandler);
+
+		[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+		[Export ("loadFileRepresentationForContentType:openInPlace:completionHandler:")]
+		NSProgress LoadFileRepresentation (UTType contentType, bool openInPlace, ItemProviderDataCompletionHandler completionHandler);
```

Comment:
`completionHandler` has a wrong signature. You could reuse `LoadInPlaceFileRepresentationHandler` but it has one different param name:

```suggestion
NSProgress LoadFileRepresentation (UTType contentType, bool openInPlace, LoadInPlaceFileRepresentationHandler completionHandler);
```

or create a new handler with right param names:

```suggestion
NSProgress LoadFileRepresentation (UTType contentType, bool openInPlace, LoadFileRepresentationHandler completionHandler);

[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
delegate void LoadFileRepresentationHandler (NSUrl fileUrl, bool openInPlace, NSError error);
```